co-log/example/1-intro/Intro.md

```diff
@@ -0,0 +1,82 @@
+# Intro: Using `LogAction`
+
+This tutorial is an introduction to `co-log`. It contains basic examples of
+using core data types and functions.
+
+You can run this tutorial by calling the following command:
+
+```shell
+cabal new-run intro
+```
+
+## Preamble: imports and language extensions
+
+Since this is a literate haskell file, we need to specify all our language
+extensions and imports up front.
+
+```haskell
+import Colog.Core (LogAction (..), (<&), logStringStdout)
+```
+
+## Core data type
+
+`co-log` is based on the following data type:
+
+```idris
+newtype LogAction m msg = LogAction
+    { unLogAction :: msg -> m ()
+    }
+```
+
+Logging action is a function from some message of user-defined type `msg` that
+perform all logic inside some monad `m`. In `co-log` logging represented as a
+value. With approach you can modify the way you do logging by simply performing
+transformations with the value you have.
+
+Let's first look at very basic example of using `LogAction`:
+
+```haskell
+example1 :: LogAction IO String -> IO ()
+example1 logger = do
+    unLogAction logger "Example 1: First message"
+    unLogAction logger "Example 1: Second message"
+```
+
+If you want to do logging, you need to pass `LogAction` as an argument to your
```

Comment:
This is a too strict statement. Probably better to say that passing the arguments id only **one of** the options, and it is the simplest one.